controllers/replicationdestination.go

```diff
@@ -42,10 +42,21 @@ func (r *VolumeSnapshotRestoreReconciler) CreateReplicationDestination(log logr.
 		},
 	}
 
+	// get restic secret created by controller
+	dmresticSecretName := fmt.Sprintf("%s-secret", vsr.Name)
+	resticSecret := corev1.Secret{}
+	if err := r.Get(r.Context, types.NamespacedName{Namespace: r.NamespacedName.Namespace, Name: dmresticSecretName}, &resticSecret); err != nil {
+		if k8serrors.IsNotFound(err) {
+			return false, nil
+		}
+		r.Log.Error(err, fmt.Sprintf("unable to fetch restic secret %s/%s", r.NamespacedName.Namespace, dmresticSecretName))
+		return false, err
+	}
+
 	// Create ReplicationDestination in protected namespace
 	op, err := controllerutil.CreateOrUpdate(r.Context, r.Client, repDestination, func() error {
 
-		return r.buildReplicationDestination(repDestination, &vsr)
+		return r.buildReplicationDestination(repDestination, &vsr, &resticSecret)
```

Comment:
Restic secret isn't being used here. Are we planning to use it in the future?

Reply:
It's being passed to replicationDestination here to add to its spec

Reply:
I see now.. it wasn't part of the diff so I missed it.. thanks for the explanation!